pkg/process/metadata/workloadmeta/collector/process.go

```diff
@@ -100,11 +100,7 @@ func (c *Collector) run(ctx context.Context, store workloadmeta.Component, conta
 }
 
 func (c *Collector) handleContainerEvent(evt workloadmeta.EventBundle) {
-	defer func() {
```

Comment:
Didn't catch this in the past but normally workloadmeta events should be acknowledged ASAP, before processing

Reply:
In that case, should we update this to call `evt.Acknowledge()` first, instead of deferring?

Reply:
I think so, but it breaks one of the tests and I think it could be done in another PR. Given that you don't do any heavy computation it should be fine.